### ISSUE TITLE
--api not recognized (already added to args EVERYWHERE!)

### ISSUE THREAD

**Issue by ezrablake911** [AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin#259]
I keep getting the "Automatic1111 is running, but you forgot to add --api flag to the webui command flags" error. I'm running Automatic1111 on PC and trying to connect to photoshop running on Mac Ventura through the gradio link.

I've added --api to commandline args in webui-user.bat:
`set COMMANDLINE_ARGS=--api --xformers`

in webui-user.sh:
`# Commandline arguments for webui.py, for example: export COMMANDLINE_ARGS="--medvram --opt-split-attention"
export COMMANDLINE_ARGS="--api --listen --xformers"`

in webui-macos-env.sh:
`export COMMANDLINE_ARGS="--api --skip-torch-cuda-test --upcast-sampling --no-half-vae --use-cpu interrogate"`

in launch.py at line 13:
`commandline_args = os.environ.get('COMMANDLINE_ARGS', "--api")`
and at line 330:
`os.environ['COMMANDLINE_ARGS'] = "--api"`

Basically, everywhere I could find commandline_args. And I'm still getting the error. Also tried restarting both Automatic and PS a dozen times. Please help!

**Comment by Kongbaiyanmie** [comment]
Me too! need. help !

**Comment by RealTekHousE** [comment]
Same here.. Not working. --api added to the .bat

**Comment by RealTekHousE** [comment]
Update, I added it to my Launch.py in one location, Line 13

`commandline_args = os.environ.get('COMMANDLINE_ARGS', "--api")`

All working now.

**Comment by dhonta** [comment]
Same problem here with this setting : 

I've added --api to commandline args in webui-user.bat:
set COMMANDLINE_ARGS=--api

in launch.py at line 13:
commandline_args = os.environ.get('COMMANDLINE_ARGS', "--api")

Stable diffusion is launch on my PC (windows10) . it seems that the plugin cannot communicate with Automatic1111.

Has anyone managed to solve this problem? Everything was working fine a week ago.

![Autophotoshop](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/34790285/7b5ed2af-c933-485f-8d3d-be664bb5c70b)

**Comment by dhonta** [comment]
I solve it by installing V1.24 with zip file. It's working now. I only have this problem with the master branch.

**Comment by ZhelenZ** [comment]
updated ✅ 
add --api everywhere ✅
not working😞

**Comment by AbdullahAlfaraj** [comment]
if you have an issue with the api:
visit this link: http://127.0.0.1:7860/sdapi/v1/cmd-flags
and screenshot the result. 
![image](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/7842232/8596a2ad-6953-49b7-93a3-ee4d0ca22021)

**Comment by dhonta** [comment]
I hope it can help : 
![api](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/34790285/9d0e24f2-226f-4b8f-8e7c-2b71afee1d8e)

**Comment by AbdullahAlfaraj** [comment]
@dhonta 
> The problem is still only with the master branch.

The issue is that the master branch contains TypeScript code that needs to be converted to JavaScript. You can either do this yourself by following the instructions in method 3 or use the zip file provided in the latest Release, where I’ve already done the conversion for you.

![image](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/7842232/2f327633-d9d3-4b9b-a446-69df491fd5de)

**Comment by dhonta** [comment]
Yes i use now the latest release and it's working fine.
Thanks for your help and your amazing work.

**Comment by ZhelenZ** [comment]
> 

{"detail":"Not Found"}

**Comment by ZhelenZ** [comment]
> if you have an issue with the api: visit this link: http://127.0.0.1:7860/sdapi/v1/cmd-flags and screenshot the result. ![image](https://user-images.githubusercontent.com/7842232/241413926-8596a2ad-6953-49b7-93a3-ee4d0ca22021.png)

{"detail":"Not Found"}

**Comment by AbdullahAlfaraj** [comment]
@ZhelenZ  if this link http://127.0.0.1:7860/docs also gives you a ‘detail not found’ error, it means you have a version of Auto1111 with a broken API.
I would suggest you open an issue in the Auto1111 repository as they will be able to help you better. https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/new/choose

**Comment by ZhelenZ** [comment]
> @ZhelenZ if this link http://127.0.0.1:7860/docs also gives you a ‘detail not found’ error, it means you have a version of Auto1111 with a broken API. I would suggest you open an issue in the Auto1111 repository as they will be able to help you better. https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues/new/choose

Thanks for your reply!
Here is what http://127.0.0.1:7860/docs gives me:
![screencapture-127-0-0-1-7860-docs-2023-05-30-11_15_46](https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/3643299/c0d72e9a-6b08-486c-8558-e43f9ab8b268)

I was able to use this Plug-in in Photoshop before I updated my Stable-diffusion-webui, and the API page showed me the Auto-Photoshop-SD part ( which couldn't be seen now).

**Comment by ZhelenZ** [comment]
> 
OS：Ventura 13.4 (M2)
AUTO1111 version:1.2.1 （not the very latest now...)
I downloaded the zip file from https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/releases/tag/v1.2.4
<img width="936" alt="Screenshot 2023-05-31 at 18 34 33" src="https://github.com/AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin/assets/3643299/61362104-15ec-4195-9c77-0a207081d5a6">

**Comment by AbdullahAlfaraj** [comment]
@ZhelenZ 
I don't recommend changing the code, as this would make it harder for you to update using ```git pull```
In your ```webui-user.sh``` you forgot to remove ```#``` before ```export COMMANDLINE_ARGS``` to uncommented.
You should properly update to Auto1111 to a new version if that doesn't work.

**Comment by meggyAI** [comment]
> 

I am getting {"detail":"Not Found"} on both links you provided. Does that mean I need to reinstall my Automatic 1111?

**Comment by AbdullahAlfaraj** [comment]
> I am getting {"detail":"Not Found"} on both links you provided. Does that mean I need to reinstall my Automatic 1111?

It's not necessary to do complete reinstallation. You Just need to upgrade.

**Comment by ZhelenZ** [comment]
> @ZhelenZ I don't recommend changing the code, as this would make it harder for you to update using `git pull` In your `webui-user.sh` you forgot to remove `#` before `export COMMANDLINE_ARGS` to uncommented. You should properly update to Auto1111 to a new version if that doesn't work.

I tried what you said. Removing "#" in "webui.sh" gives me errors  (fixed by putting "#" back); upgrade the Auto1111 didn't help either.

But thank you anyway ~

**Comment by xgt1986627** [comment]
I also have this problem. Is it possible that the latest version is not better for V1.2.4 you mentioned?
